include/ipr/interface

```diff
@@ -1526,6 +1548,13 @@ namespace ipr {
    // An expression of the form "static_cast<type>(expr)".
    struct Static_cast : Cast_expr<static_cast_cat> { };
 
+                                   // -- Qualification --
+   // Implicit conversions that add cv-qualifiers
+   struct Qualification : Binary<Category<qualified_cat>, const Type&> {
+      Arg1_type target() const { return first(); }
```

Comment:
I suggest we use the notation where the source expression comes first, then the qualifier comes second:

```
struct Qualification : Binary<Category<qualification_cat>, const Expr&, Type_qualifier> {
   Arg1_type expr() const { return first(); }
   Arg2_type qualifier() const { return second(); }
};
```